pkgs/applications/science/logic/surelog/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, python3
+, pkg-config
+, libuuid
+, openjdk11
+, gperftools
+}:
+
+stdenv.mkDerivation rec {
+  pname = "Surelog";
+  version = "2022.05.15";
```

Comment:
For Version number: Surelog actually has a built-in version number
(https://github.com/chipsalliance/Surelog/blob/120a9d89001c8ecc2c63fdabee56538476c6f9f7/src/CommandLine/CommandLineParser.cpp#L66 ), which gets advanced every now and then.

..and that is what I am using in #185626 (which I'll abandon if we can get your change submitted).

Unfortunately, it is not tagged there in github yet with that version number, so currently we have to manually get it out of there, but I hope that tagging happens upstream soon.

So in order to avoid some weird version jump later (2022.xx.yy -> 1.36, which is a weird ordering as 2022 >> 1.), maybe we should now start using that version ? Meanwhile I attempt to get upstream to add the tags, so that it might make things simpler here in the future.

It might also not be problematic at all to have weird version changes, so we can also wait until that upstream tagging is happening.

Up to you.

Reply:
I don't have time today to do a more thorough review but, yes: if there are already some built-in tags that get advanced every now and then, let's use that, and try to encourage upstream to update more frequently as needed, rather than using random dates. In particular it makes the support/QA process much simpler to have aligned version numbers.

If we need to switch version schemes I don't think it's that terrible (it kind of messes with `nix-env`'s `compareVerions` path, but that's normally OK) but it would be nice to get it right the first time.

Reply:
I now got upstream to create a tag so the following should now work:

```
  pname = "surelog";
  version = "v1.36";

  src = fetchFromGitHub {
    owner = "chipsalliance";
    repo = "Surelog";
    rev = version;
    hash = "sha256-gMxpnSsJTXmg2VU7mQcw1J40OuqW6oo9/sGplLgV0ho=";
    fetchSubmodules = true;
  };
```

Reply:
w.r.t. the name: I prefer a canonical lowercase name for the derivation and just use the uppercase for the git repo name. But I leave that up to whatever style is preferred.

Reply:
Yes, the name should be lowercased. Note that GitHub is actually case-insensitive when doing clones, so simply setting `pname = "surelog";` and then `fetchFromGitHub { repo = pname; }` will work fine and you don't need to duplicate the two, but honestly I don't think that duplication matters much, the repository name is extremely unlikely to change.